sklearn/base.py

```diff
@@ -150,8 +150,14 @@ class BaseEstimator:
     arguments (no ``*args`` or ``**kwargs``).
     """
 
+    _param_names_cache = dict()
```

Comment:
```suggestion
    _param_names_cache = {}
```

Reply:
I've seen other parts in the code use dict(), but whatever you prefer.